homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +85,53 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+# Binary sensor service map
+BINARY_SENSOR_SERVICE_MAP = {
+    DEVICE_CLASS_CO2:
+        (SERV_CARBON_DIOXIDE_SENSOR, CHAR_CARBON_DIOXIDE_DETECTED),
+    DEVICE_CLASS_GAS:
+        (SERV_CARBON_MONOXIDE_SENSOR, CHAR_CARBON_MONOXIDE_DETECTED),
+    DEVICE_CLASS_MOISTURE:
+        (SERV_LEAK_SENSOR, CHAR_LEAK_DETECTED),
+    DEVICE_CLASS_MOTION:
+        (SERV_MOTION_SENSOR, CHAR_MOTION_DETECTED),
+    DEVICE_CLASS_OCCUPANCY:
+        (SERV_OCCUPANCY_SENSOR, CHAR_OCCUPANCY_DETECTED),
+    DEVICE_CLASS_OPENING:
+        (SERV_CONTACT_SENSOR, CHAR_CONTACT_SENSOR_STATE),
+    DEVICE_CLASS_SMOKE:
+        (SERV_SMOKE_SENSOR, CHAR_SMOKE_DETECTED)}
+
+
+@TYPES.register('BinarySensor')
+class BinarySensor(HomeAccessory):
+    """Generate a BinarySensor accessory as binary sensor."""
+
+    def __init__(self, hass, entity_id, name, **kwargs):
+        """Initialize a BinarySensor accessory object."""
+        super().__init__(name, entity_id, CATEGORY_SENSOR, **kwargs)
+
+        self.hass = hass
+        self.entity_id = entity_id
+
+        attributes = hass.states.get(entity_id).attributes
+        device_class = attributes.get(ATTR_DEVICE_CLASS)
+        service_char = BINARY_SENSOR_SERVICE_MAP[device_class] \
+            if (device_class in BINARY_SENSOR_SERVICE_MAP) \
```

Comment:
You don't need brackets here. `if device_class in BINARY_SENSOR_SERVICE_MAP \` is enough.